tapchannel/auf_leaf_signer_test.go

```diff
@@ -131,6 +142,171 @@ func TestAuxLeafSigner(t *testing.T) {
 	}
 }
 
+// TestAuxLeafSignerCancel tests that the AuxLeafSigner will handle a cancel
+// signal correctly, which involves skipping all remaining sig jobs.
+func TestAuxLeafSignerCancel(t *testing.T) {
+	cfg := &LeafSignerConfig{
+		ChainParams: testChainParams,
+		Signer:      &mockVirtualSigner{},
+	}
+
+	signer := NewAuxLeafSigner(cfg)
+	require.NoError(t, signer.Start())
+
+	defer func() {
+		require.NoError(t, signer.Stop())
+	}()
+
+	chanState := &channeldb.OpenChannel{
+		ChanType: channeldb.AnchorOutputsBit |
+			channeldb.ScidAliasChanBit | channeldb.SingleFunderBit |
+			channeldb.SimpleTaprootFeatureBit |
+			channeldb.TapscriptRootBit,
+		IsInitiator: true,
+	}
+	randInputProof := randProof(t)
+	commitTx := &randInputProof.AnchorTx
+	outgoingHtlcs := make(map[input.HtlcIndex][]*cmsg.AssetOutput)
+	outgoingHtlcs[0] = []*cmsg.AssetOutput{
+		cmsg.NewAssetOutput(
+			randInputProof.Asset.ID(), randInputProof.Asset.Amount,
+			randInputProof,
+		),
+	}
+
+	com := cmsg.NewCommitment(
+		nil, nil, outgoingHtlcs, nil, lnwallet.CommitAuxLeaves{},
+	)
+	cancelChan := make(chan struct{})
+
+	// Constructing multiple jobs will allow us to assert that later jobs
+	// are cancelled successfully.
+	numSigJobs := 10
+	jobs := make([]lnwallet.AuxSigJob, 0, numSigJobs)
+	outputIdxCounter := int32(0)
+
+	for range numSigJobs {
+		newJob := RandAuxSigJob(
+			t, cancelChan, lfn.Some[tlv.Blob](com.Bytes()),
+			outputIdxCounter,
+		)
+		jobs = append(jobs, newJob)
+		outputIdxCounter++
+	}
+
+	// We'll try to cancel all signature jobs once the aux signer completes
+	// a third of the batch.
+	cancelThreshold := numSigJobs / 3
+
+	err := signer.SubmitSecondLevelSigBatch(chanState, commitTx, jobs)
+	require.NoError(t, err)
+
+	select {
+	case <-time.After(testTimeout):
+		t.Fatalf("timeout waiting for response")
+	case <-jobs[cancelThreshold].Resp:
+		// Send the cancel signal; jobs at the end of the batch should
+		// not be processed.
+		close(cancelChan)
```

Comment:
Won't this always lead to a panic in the test? As `cancelChan` will be closed more than once. 

As mentioned earlier, perhaps what we want is to wrap it in some light abstraction over a `sync.Once`?